src/frequenz/sdk/microgrid/_power_managing/_power_managing_actor.py

```diff
@@ -84,9 +78,17 @@ def __init__(  # pylint: disable=too-many-arguments
         self._bound_tracker_tasks: dict[frozenset[int], asyncio.Task[None]] = {}
         self._subscriptions: dict[frozenset[int], dict[int, Sender[_Report]]] = {}
 
-        self._algorithm: BaseAlgorithm = Matryoshka(
-            max_proposal_age=timedelta(seconds=60.0)
-        )
+        match algorithm:
+            case Algorithm.MATRYOSHKA:
+                self._algorithm: BaseAlgorithm = Matryoshka(
+                    max_proposal_age=timedelta(seconds=60.0)
+                )
+            case Algorithm.SHIFTING_MATRYOSHKA:
+                self._algorithm = ShiftingMatryoshka(
+                    max_proposal_age=timedelta(seconds=60.0)
+                )
+            case _:
+                assert_never(algorithm)
```

Comment:
This change is in the commit that says it is changing the default algorithm, I would move it into the commit adding the new algorithm instead.